### PR TITLE
Zachmu/triggers bugfix

### PR DESCRIPTION
**Pull request by zachmu**
Fixed bug in insert triggers, which couldn't handle out-of-order column insertions.

Fixes https://github.com/dolthub/dolt/issues/950

### CODE REVIEW

File: enginetest/memory_engine_test.go [hunk 1]
```diff
@@ -100,34 +100,38 @@ func TestSingleScript(t *testing.T) {
 
 	var test enginetest.ScriptTest
 	test = enginetest.ScriptTest{
-		Name: "multiple triggers before and after insert, with precedes / follows",
+		Name: "trigger before insert, multiple triggers defined",
```
Comment (andrew-wm-arthur) [inline enginetest/memory_engine_test.go:103]: this should go in `enginetest/trigger_queries.go`

File: sql/plan/trigger.go [hunk 2]
```diff
@@ -93,14 +93,42 @@ type triggerIter struct {
...
+			return &prependNode{
+				UnaryNode: UnaryNode{Child: n},
+				row:       row,
+			}, nil
+		default:
+			return n, nil
+		}
+	}
+}
+
```
Comment (andrew-wm-arthur) [inline sql/plan/trigger.go:123]: what's the difference between 
1) conditionally prepending the child row, as you do here
2) always prepending, and manipulating the parent expressions to access the correct fields of the composite row

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM. 
Tests looks good, one comment about implementation

File: sql/plan/trigger.go (hunk 2 above)
Comment (zachmu) [reply sql/plan/trigger.go:123]: They're coupled. The parent expressions get their indexes from the analyzer using the scope, which is basically a promise that we'll fill in exactly the right number of outers scope columns at the beginning of the row at runtime. The issue is that for inserts, we don't ever consider the scope value, so we can't prepend any values to the projection.

This is complicated and there are still probably bugs in this approach.

File: enginetest/memory_engine_test.go (hunk 1 above)
Comment (zachmu) [reply enginetest/memory_engine_test.go:103]: It is, I just copy and paste tests there when I want to run them in isolation. It causes churn in this file, but I'm not super concerned.

**Pull request merged**